spaceship/lib/spaceship/portal/portal_client.rb

```diff
@@ -110,20 +151,40 @@ def platform_slug(mac)
     # @!group Apps
     #####################################################
 
+    # <b>DEPRECATED:</b> Use <tt>apps_by_platform</tt> instead.
     def apps(mac: false)
-      paging do |page_number|
-        r = request(:post, "account/#{platform_slug(mac)}/identifiers/listAppIds.action", {
-          teamId: team_id,
-          pageNumber: page_number,
-          pageSize: page_size,
-          sort: 'name=asc'
-        })
-        parse_response(r, 'appIds')
+      Helper.log.warn '`apps` is deprecated. Please use `apps_by_platform` instead.'.red
+      apps_by_platform(platform: mac ? Spaceship::Portal::App::MAC : Spaceship::Portal::App::IOS)
+    end
+
+    def apps_by_platform(platform: nil)
+      platforms = [platform]
+      platforms = Spaceship::Portal::App::PLATFORMS if platform.nil?
+
+      results = []
+
+      platforms.each do |the_platform|
+        output = paging do |page_number|
+          r = request(:post, Spaceship::PortalConstants::APP_ID_URL[the_platform], {
+              teamId: team_id,
+              pageNumber: page_number,
+              pageSize: page_size,
+              sort: 'name=asc'
+          })
+          parse_response(r, Spaceship::PortalConstants::EXPECTED_KEY_APP_ID_URL[the_platform])
+        end
+
+        output.each {|app| app['appIdPlatform'] = the_platform } if Spaceship::Portal::App::ADD_PLATFORM.include? the_platform # We add the platform here for some of the app platform types because there response data does not include it.
+        results += output
       end
+
+      results
     end
 
     def details_for_app(app)
-      r = request(:post, "account/#{platform_slug(app.mac?)}/identifiers/getAppIdDetail.action", {
+      raise 'The developer portal does not allow details the specified platform' if app.platform == Spaceship::Portal::App::MERCHANT or app.platform == Spaceship::Portal::App::ICLOUD or app.platform == Spaceship::Portal::App::PASS or app.platform == Spaceship::Portal::App::WEB
```

Comment:
This line is a bit long as well 😄  What do you think about pulling that conditional check into a helper method?

```ruby
def can_request_details_for(platform)
	![
	  Spaceship::Portal::App::MERCHANT,
	  Spaceship::Portal::App::ICLOUD,
	  Spaceship::Portal::App::PASS,
	  Spaceship::Portal::App::WEB
	].include?(platform)
end
```

and then use it as:

```ruby
raise "The developer portal does not allow details requests for platform: #{app.platform}" unless can_request_details_for(app.platform)
```